### DISCUSSION TITLE
Alternate link providers parsing

### DISCUSSION THREAD

**Discussion by CoolnsX** [pystardust/ani-cli#685] [Ideas] [answered]
@port19x see this..
![Screenshot_20220424-012413-882](https://user-images.githubusercontent.com/76195824/164958265-e32d08ed-cec1-4338-9b63-017dd65bc518.png)

Download testing..
xstreamcdn > MP4 upload > goload > streamsb (m3u8) > doodstream
I m trying to figure out bypassing google recaptcha on download page.. in this way we get those gogocdn (MP4) links along with streamsb (MP4) links cause streamsb also provides mulitquality highspeed links

**Comment by CoolnsX** [comment, marked as answer]
Here is the script..
[link](https://0x0.st/oTpq.sh) 
I modified it little bit.. its scraping the front page..
Directly run it.. and select any episode from list..
And see the results..

**Comment by port19x** [comment]
Looks promising

**Reply by CoolnsX** [reply]
> Looks promising

Providers infos..
Xstreamcdn (multi - quality high speed MP4 links)
mp4upload ( single quality high speed MP4 link ) 
doodstream ( single quality MP4 link )
goload ( m3u8 links )
streamsb ( m3u8 links - only for downloading, cause I m unable to stream those links, even though I passed the same headers )

Also I m thinking of using m3u8 links for streaming and MP4 links for downloading..

For downloading:-
Xstreamcdn > mp4upload > doodstream > gogocdn

For streaming:-
Goload > doodstream > mp4upload > xstreamcdn

Or another way is to use them randomly..

**Reply by CoolnsX** [reply]
And I don't wanna put all the pressure on best provider.. cause we hv done it in the past and gogo put the captcha on it.. so that's why this reply..
